tests/test_ec2.py

```diff
@@ -205,6 +205,24 @@ def test_ec2_resize(self):
             list(sorted(['m4.large', 'm4.2xlarge', 'm4.2xlarge'])))
 
 
+class TestStateTransitionAgeFilter(BaseTest):
+
+    def test_ec2_state_transition_age(self):
+        session_factory = self.replay_flight_data(
+            'test_ec2_state_transition_age_filter'
+        )
+        policy = self.load_policy({
+            'name': 'ec2-state-transition-age',
+            'resource': 'ec2',
+            'filters': [
+                {'State.Name': 'running'},
+                {'type': 'state-age',
+                 'days': 30}]},
+            session_factory=session_factory)
+        resources = policy.run()
+        self.assertEqual(len(resources), 1)
```

Comment:
I'll add that in. As for why only one instance matches the filter: while there are two instances whose state transition ages are greater than the value this tests for, one of them isn't running so it gets excluded by the state name filter. I can take out the state name filter if you want, I can see how it could obscure the intent of the test.